File: Baekjoon/Swift/Implementation/Q21312.swift
//
//  main.swift
//  BOJ_PS
//
//  Created by 김태훈 on 2021/05/11.
//  홀짝 칵테일 / 구현

let beverage = readLine()!.split(separator: " ").map { Int(String($0))! }
var isOdd = false
var res = 1

func solution() {
    for i in beverage.indices {
        if beverage[i] % 2 == 1 {
            res *= beverage[i]
            isOdd = true
        }
    }
    
    print(isOdd == true ? res : beverage.reduce(1,*))
}

solution()

File: Baekjoon/Python/BOJ_PS/Implementation/Q21312.py
# 홀짝 칵테일 / 구현
beverages = list(map(int, input().split()))
res = 1
isOdd = False
for beverage in beverages:
    if beverage % 2 == 1:
        res *= beverage
        isOdd = True

print(res if isOdd == True else beverages[0]*beverages[1]*beverages[2])